lib/ansible/modules/remote_management/imc/imc_rest.py

```diff
@@ -408,6 +408,7 @@ def main():
         # Perform actual request
         resp, info = fetch_url(module, url, data=data, method='POST', timeout=timeout)
         if resp is None or info['status'] != 200:
+            logout(module, url, cookie, timeout)
```

Comment:
This shouldn't be needed, as `atexit` takes care of it. In fact, `atexit` also takes care of unexpected failures by logging out properly.